app/models/report_query.rb

```diff
@@ -23,7 +23,7 @@ class ReportQuery < ActiveRecord::Base
   validates_uniqueness_of :name, :scope => :collection_id
 
   belongs_to :collection
-  has_many :report_query_templates, dependent: :restrict
+  has_many :report_query_templates, dependent: :destroy
```

Comment:
this case will allow a user to destroy report query even this report query has some report query template references, we should find out another way to deal with remove collection activity